test/integ/cli/test_cli_attachment.py

```diff
@@ -0,0 +1,285 @@
+        assert json.loads(result.output)["processed_bytes"] == TEST_FILE_SIZE
+
+    @pytest.mark.parametrize("manifest_case_key", MOCK_MANIFEST_CASE.keys())
+    def test_attachment_path_mapping_flow(
+        self, temp_dir, job_attachment_resources, manifest_case_key
+    ):
+
+        # Given
+        source_path: str = os.path.join(temp_dir, "virtual_source")
+        destination_path: str = temp_dir
+
+        file_name: str = f"{hash_data(source_path.encode('utf-8'), HashAlgorithm.XXH128)}_output"
+        manifest_path: str = os.path.join(temp_dir, file_name)
+
+        with open(
+            manifest_path,
+            "w",
+            encoding="utf8",
+        ) as f:
+            json.dump(MOCK_MANIFEST_CASE[manifest_case_key], f)
+
+        path_mapping_file_path: str = os.path.join(temp_dir, "path_mapping")
+        with open(path_mapping_file_path, "w", encoding="utf8") as f:
+            f.write(
+                json.dumps(
+                    [
+                        asdict(
+                            PathMappingRule(
+                                source_path_format="posix",
+                                source_path=source_path,
+                                destination_path=destination_path,
+                            )
+                        )
+                    ]
+                )
+            )
+
+        s3_root_uri = f"s3://{job_attachment_resources.bucket_name}/{job_attachment_resources.bucket_root_prefix}"
+
+        runner = CliRunner()
+        # Temporary, always add cli_attachment until launched.
+        main.add_command(cli_attachment)
+
+        # When - test upload the local asset file with path mapping
+        result = runner.invoke(
+            main,
+            [
+                "attachment",
+                "upload",
+                "--manifests",
+                manifest_path,
+                "--path-mapping-rules",
+                path_mapping_file_path,
+                "--profile",
+                "default",
+                "--s3-root-uri",
+                s3_root_uri,
+            ],
+        )
+        # Then
+        assert result.exit_code == 0, f"Non-Zeo exit code, CLI output {result.output}"
+
+        # When - test download the file just uploaded with path mapping
+        result = runner.invoke(
+            main,
+            [
+                "attachment",
+                "download",
+                "--manifests",
+                manifest_path,
+                "--path-mapping-rules",
+                path_mapping_file_path,
+                "--profile",
+                "default",
+                "--s3-root-uri",
+                s3_root_uri,
+                "--json",
+            ],
+        )
+        # Then
+        assert result.exit_code == 0, f"Non-Zeo exit code, CLI output {result.output}"
+        assert json.loads(result.output)["processed_bytes"] == TEST_FILE_SIZE
```

Comment:
Should we check that files are downloaded? EG: stat the content of the file ?

Reply:
Added check for file name downloaded, and updated the check on processed bytes.